src/xo-app/backup/restore/index.js

```diff
@@ -98,16 +103,88 @@ const doImport = ({ backup, sr, start }) => {
   }
 }
 
+class _Collapsible extends Component {
+  render () {
+    const {
+      collapsible,
+      children,
+      ...props
+    } = this.props
+
+    return collapsible
+      ? <Collapse {...props}>{children}</Collapse>
+      : <div>
+        <span>{props.buttonText}</span>
+        <br />
+        {children}
+      </div>
+  }
+}
+
 class _ModalBody extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      mapVdisSrs: {}
+    }
+  }
+
   get value () {
     return this.state
   }
 
+  _getSrPredicate = createSelector(
+    () => this.state.sr,
+    () => this.state.mapVdisSrs,
+    (defaultSr, mapVdisSrs) => sr => {
+      let result = isSrWritable(sr) && defaultSr.$pool === sr.$pool
+
+      if (!defaultSr.shared) {
+        result = result && (sr.shared || defaultSr.$container === sr.$container)
```

Comment:
What I meant is you can test `sr.shared` even before this whole `if` section if it's a global reason for a SR to be accepted.